hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/UpdateStrategy.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.api.graph;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.validation.constraints.NotNull;
+
+import com.baidu.hugegraph.exception.NotSupportException;
+import com.baidu.hugegraph.util.E;
+import com.google.common.collect.Sets;
+
+public enum UpdateStrategy {
+
+    //Only number support accumlate
+    ACCUMULATE {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            // TODO: How to accumulae universally (Byte, Long,...Double)
+            return (Double) oldProperty + (Double) newProperty;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            E.checkState(oldProperty instanceof Number &&
+                         newProperty instanceof Number,
+                         getFormat(oldProperty, newProperty, "Number"));
+        }
+    },
+
+    // Only Date/Number support bigger/smaller
+    // TODO: Support Number compare later
+    BIGGER {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return getComparedDate(oldProperty, newProperty, BIGGER);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            E.checkState((oldProperty instanceof Date ||
+                          oldProperty instanceof Long) &&
```

Comment:
oldProperty instanceof Number?

Reply:
have thought about it before, but the timestamp with milliseconds can only be Long, the unattached is also close to the upper limit of Integer, and it should only allow Integer or Long?  The range of Number seems too wide

Reply:
Maybe we can use method compareNumber from hugegraph-common directly: `com.baidu.hugegraph.util.NumericUtil.compareNumber(Object first, Number second)` 
and call `com.baidu.hugegraph.util.NumericUtil.convertToNumber(Object value)` to convert an object to a number.